packages/gatsby-plugin-sharp/src/index.js

```diff
@@ -363,6 +363,13 @@ async function responsiveSizes({ file, args = {} }) {
     options.sizes = `(max-width: ${options.maxWidth}px) 100vw, ${options.maxWidth}px`
   }
 
+  // Account for images with a high pixel density. We assume that these types
+  // of images are intended to be used as high resolution ("retina") images.
```

Comment:
Perhaps change the comment to:

"We assume that these types of images are intended to be displayed at their native resolution."

The sentence as it reads now is confusing because we always aim to provide users with retina-class devices a high resolution image. The fix is to avoid expanding the images themselves beyond their original width/height.

Reply:
Good point. Updated the comment!